src/main.lua

```diff
@@ -83,9 +83,9 @@ local ActionMap = {
 Handlers.add(ActionMap.Transfer, utils.hasMatchingTag("Action", ActionMap.Transfer), function(msg)
 	-- assert recipient is a valid arweave address
 	local function checkAssertions()
-		assert(utils.isValidArweaveAddress(msg.Tags.Recipient), "Invalid recipient")
+		assert(utils.isValidAOAddress(msg.Recipient), "Invalid recipient")
 		assert(
-			tonumber(msg.Tags.Quantity) > 0 and utils.isInteger(tonumber(msg.Tags.Quantity)),
+			tonumber(msg.Quantity) > 0 and utils.isInteger(tonumber(msg.Quantity)),
```

Comment:
can you confirm Quantity is a higher level attribute on all messages?

Reply:
I'm actually a bit confused on when values should be read from msg.Tags and when they're read from msg directly. I noticed in the token blueprint it is reading Quantity and Recipient directly from msg:

https://cookbook_ao.g8way.io/guides/aos/blueprints/token.html

Reply:
I believe AO infra will append specific tags as higher-level attributes to all messages. I believe they do so for `Quantity`, `From`, `Recepient` but it doesn't show anywhere in the spec - https://cookbook_ao.ar-io.dev/concepts/messages.html

hence why I've always deferred to using Tags as that's what they are on the actual data item

Ref: https://ao.ar.io/#/read

<img width="717" alt="image" src="https://github.com/user-attachments/assets/e7e220d0-ac97-487e-b86e-ad49d7e11ee7">

Reply:
I still find it ambiguous, but to be safe I've normalized back to msg.Tags.Quantity and msg.Tags.Recipient